hyperspy/model.py

```diff
@@ -1257,11 +790,10 @@ def fit(self, fitter=None, method='ls', grad=False,
         elif fitter == "odr":
             modelo = odr.Model(fcn=self._function4odr,
                                fjacb=odr_jacobian)
-            mydata = odr.RealData(
-                self.axis.axis[
-                    self.channel_switches], self.spectrum()[
-                    self.channel_switches], sx=None, sy=(
-                    1 / weights if weights is not None else None))
+            mydata = odr.RealData(self.axis.axis[self.channel_switches],
```

Comment:
it also doesn't have self._jacobian so I don't think the odr fitter will work for many reasons at the moment... My thinking at the moment was lets separate out as much as we can that can be general compared to that which can't and then go back through and add in functionality for Model2D bit-by-bit. Mostly because if we tried to do everything in one go the diff would be massive and it would be hard to track down errors.

I actually want to use the gradient based fitting methods in 2D for something I'm working on so this will be addressed before long.